### PR TITLE
Mark WebTransport streams keep alive

### LINKED ISSUE #1369: Mark WebTransport streams keep-alive
We need to propagate neqo_transport::Connection to  this function and call this function:
 https://github.com/mozilla/neqo/blob/25a862e97072a06b2ac93a6d5035e0202c0c7d6f/neqo-http3/src/connection.rs#L1048

### PR DESCRIPTION
**Pull request by MayyaSunil**
This PR is intended to resolve #1369

### CODE REVIEW

**Review by martinthomson — CHANGES_REQUESTED** [review]
See my comments on the issue.

**Review by martinthomson — CHANGES_REQUESTED** [review]
I don't think you have the right places for these calls.

You need one for the server and one for the client.

I think that the client one goes [in webtransport_create_session](https://github.com/mozilla/neqo/blob/b045c4a7033feffb133d05f86acea74dc50084ec/neqo-http3/src/connection.rs#L1054) and the server one goes [in webtransport_accept_session](https://github.com/mozilla/neqo/blob/b045c4a7033feffb133d05f86acea74dc50084ec/neqo-http3/src/connection.rs#L1098).

The points you have identified are about streams created within the session, not the session stream.

Finally, I would like to see test cases for this, showing that keep-alives (PING) are sent if the connection is left to go idle for too long, with only a webtransport session active.  There are test cases for keep-alive that you can base this on.

**Comment by larseggert** [comment]
@MayyaSunil are you still intending to land this change?

**Comment by larseggert** [comment]
@MayyaSunil any update on when you might get to this?

**Comment by MayyaSunil** [comment]
> @MayyaSunil any update on when you might get to this?

I will resume this early next week and hope to complete by the end of the week

**Comment by larseggert** [comment]
OK. Making this a draft PR until it's ready for review, please change it back then.

**Comment by larseggert** [comment]
@MayyaSunil is this good to go functionality-wise? Does it still need a test?

**Comment by MayyaSunil** [comment]
@larseggert Its not okay functionality wise and needs a test

**Comment by larseggert** [comment]
CC @jesup

**Comment by larseggert** [comment]
Webtransport newbie here. Trying to help wrap this up.

> I think that the client one goes [in webtransport_create_session](https://github.com/mozilla/neqo/blob/b045c4a7033feffb133d05f86acea74dc50084ec/neqo-http3/src/connection.rs#L1054)

The call to `create_bidi_transport_stream` in that function already enabled keepalives. Hence, are we done for the client?

> and the server one goes [in webtransport_accept_session](https://github.com/mozilla/neqo/blob/b045c4a7033feffb133d05f86acea74dc50084ec/neqo-http3/src/connection.rs#L1098).

I've made that addition, and added a test. WDYT?

**Review by KershawChang — APPROVED** [review]
This looks good to me.
Thanks!

File: neqo-http3/tests/webtransport.rs [hunk 1]
```diff
@@ -229,6 +229,15 @@ fn receive_data_server(
     wt_stream.unwrap()
 }
 
+#[test]
+fn wt_keepalive() {
+    let (mut client, mut server) = connect();
+    let _wt_session = create_wt_session(&mut client, &mut server);
+    let idle_timeout = ConnectionParameters::default().get_idle_timeout();
+    assert_eq!(client.process_output(now()).callback(), idle_timeout / 2);
+    assert_eq!(server.process_output(now()).callback(), idle_timeout / 2);
```
Comment (mxinden) [inline neqo-http3/tests/webtransport.rs:238]:
```suggestion
    // Expect client and server to send PING after half of the idle timeout in order to keep connection alive.
    assert_eq!(client.process_output(now()).callback(), idle_timeout / 2);
    assert_eq!(server.process_output(now()).callback(), idle_timeout / 2);
```

Feel free to ignore.

**Review by mxinden — APPROVED** [review]
For what my review is worth, this looks good to me.

**Comment by larseggert** [comment]
@martinthomson could you clear your old review, or better yet re-review?

**Review by martinthomson — APPROVED** [review]
This is certainly a lot cleaner than I remember it being.

**Pull request merged**